awx/main/scheduler/task_manager_models.py

```diff
@@ -34,14 +34,14 @@ def remaining_capacity(self):
 
 
 class TaskManagerInstances:
-    def __init__(self, active_tasks, instances=None):
+    def __init__(self, active_tasks, instances=None, instance_fields=['node_type', 'capacity', 'hostname', 'enabled']):
         self.instances_by_hostname = dict()
+        self.instance_objects = []
         if instances is None:
-            instances = (
-                Instance.objects.filter(hostname__isnull=False, enabled=True).exclude(node_type='hop').only('node_type', 'capacity', 'hostname', 'enabled')
-            )
+            instances = Instance.objects.filter(hostname__isnull=False, enabled=True).exclude(node_type='hop').only(*instance_fields)
         for instance in instances:
             self.instances_by_hostname[instance.hostname] = TaskManagerInstance(instance)
+            self.instance_objects.append(instance)
```

Comment:
I don't understand why you need to introduce a new data structure. You could just loop over `self.instances_by_hostname.values()` and for each item reference `.obj`.

This is a change I'd like to get in here, because this objects are functionally important to the task manager itself, and I don't like the idea of modifying them just for the collectors. And I think getting rid of this will actually make the collector code more readable.